src/gei/Commands/MigrateArchiveRepoCommand.cs

```diff
@@ -0,0 +1,195 @@
+using System.IO;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class MigrateArchiveRepoCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+        private readonly IAzureApiFactory _azureApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        private const int ARCHIVE_GENERATION_TIMEOUT_IN_HOURS = 10;
+        private const int CHECK_STATUS_DELAY_IN_MILLISECONDS = 10000; // 10 seconds
+        private const string GIT_ARCHIVE_FILE_NAME = "gitArchive.tar.gz";
+        private const string METADATA_ARCHIVE_FILE_NAME = "metadataArchive.tar.gz";
+
+        public MigrateArchiveRepoCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, ITargetGithubApiFactory targetGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider, IAzureApiFactory azureApiFactory) : base("migrate-archive-repo")
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _azureApiFactory = azureApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+            _targetGithubApiFactory = targetGithubApiFactory;
+
+            Description = "Generates migration archives, uploads them to Azure Blob Storage, then invokes the GitHub Migration APIs to migrate the repo and all repo data using those uploaded archives.";
+            Description += Environment.NewLine;
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT is optional, if not set GH_PAT will be used instead. This authenticates to the source GHES API.";
+
+            var ghesApiUrl = new Option<string>("--ghes-api-url")
+            {
+                IsRequired = true,
+                Description = "The api endpoint for the hostname of your GHES instance. For example: https://api.myghes.com"
+            };
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true
+            };
+            var sourceRepo = new Option<string>("--source-repo")
+            {
+                IsRequired = true
+            };
+            var githubTargetOrg = new Option<string>("--github-target-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_PAT env variable."
+            };
+            var targetRepo = new Option<string>("--target-repo")
+            {
+                IsRequired = false,
+                Description = "Defaults to the name of source-repo"
+            };
+            var azureStorageConnectionString = new Option<string>("--azure-storage-connection-string")
+            {
+                IsRequired = true,
```

Comment:
I think this should be false, if we let them pass it in via env vars

Reply:
Rogan said the CLI arg should be preferred (that we should expect customers know nothing)

But updated in https://github.com/github/gh-gei/pull/221/commits/8f82a50fc9abfa8950e6d9d8d55e7b71e35f9b13.